File: app/src/main/java/com/example/baseball/domain/LifeCount.kt
package com.example.baseball.domain

data class LifeCount(private var _count : Int = 1) {
    init {
        require(_count in MIN_LIFE_COUNT..MAX_LIFE_COUNT)
    }
    val count : Int
        get() = _count

    fun increase() : Boolean {
        if(_count >= MAX_LIFE_COUNT) return false
        _count++
        return true
    }

    fun decrease() : Boolean {
        if(_count <= MIN_LIFE_COUNT) return false
        _count--
        return true
    }
    companion object {
        private const val MIN_LIFE_COUNT = 1
        private const val MAX_LIFE_COUNT = 19
    }
}

File: domain/domain/LifeCount.swift
//
//  LifeCount.swift
//  iOSbaseball
//
//  Created by KimMinSeok on 5/18/24.
//
enum LifeCountError: Error {
    case IllegalArgumentException
}

import Foundation

public class LifeCount{
    
    public private(set) var lifes : Int
    
    public init(lifes: Int = 1) throws {
        self.lifes = lifes
        
        guard (MIN_LIFE_COUNT...MAX_LIFE_COUNT).contains(self.lifes) else {
             throw LifeCountError.IllegalArgumentException
        }
    }
    
    public func increase() -> Bool{
        return lifes < MAX_LIFE_COUNT ? { lifes += 1; return true }() : false
    }
    
    public func decrease() -> Bool{
        return lifes > MIN_LIFE_COUNT ? { lifes -= 1; return true }() : false
    }
    
    private let MIN_LIFE_COUNT = 1
    private let MAX_LIFE_COUNT = 20
}
